unity/Holo/Assets/Editor/ModelImport/STLImport/STLImporter.cs

```diff
@@ -1,83 +1,83 @@
-﻿using System.Collections.Generic;
-using System.IO;
-using System.Text;
-using UnityEngine;
-
-
-//Loads a single STL file and turns it into a list of vertices (x,y,z) & if firstMesh: a list of indexes
-public class STLImporter
-{
-    private List<int> indices = new List<int>();
-    public int[] Indices { get => indices.ToArray(); }
-
-    public Vector3[] BaseVertices { get; private set; }
-
-    private List<Vector3> vertices = new List<Vector3>();
-    public Vector3[] Vertices { get => vertices.ToArray(); }
-
-    private List<Vector3> normals = new List<Vector3>();
-    public Vector3[] Normals { get => normals.ToArray(); }
-
-
-
-    private uint facetCount;
-
-    public void LoadFile(string file_path)
-    {
-        vertices.Clear();
-        indices.Clear();
-        normals.Clear();
-        using (FileStream filestream = new FileStream(file_path, FileMode.Open, FileAccess.Read))
-        {
-            using (BinaryReader binaryReader = new BinaryReader(filestream, new ASCIIEncoding()))
-            {
-                // read header
-                byte[] header = binaryReader.ReadBytes(80);
-                facetCount = binaryReader.ReadUInt32();
-
+﻿using System.Collections.Generic;
+using System.IO;
+using System.Text;
+using UnityEngine;
+
+
+//Loads a single STL file and turns it into a list of vertices (x,y,z) & if firstMesh: a list of indexes
+public class STLImporter

```

Comment:
It would be nice excercise to derive both STL and VTKImporter from common interface and use it in unified way so you can skip the distinction from extensions on higher level.